src/fu-engine.c

```diff
@@ -621,7 +621,10 @@ fu_engine_modify_remote (FuEngine *self,
 		return FALSE;
 	}
 	g_key_file_set_string (keyfile, "fwupd Remote", key, value);
-	return g_key_file_save_to_file (keyfile, filename, error);
+	g_debug ("Reloading configuration files");
+	if (!g_key_file_save_to_file (keyfile, filename, error))
+		return FALSE;
+	return fu_config_load (self->config, FU_CONFIG_LOAD_FLAG_NONE, error);
```

Comment:
ohh, I don't think it's that easy. `fu_config_load()` sets up all kind of stuff that's deallocated if it's re-used. I think you need something exported like `fu_quirks_check_silo` which uses `xb_silo_is_valid()`...

Reply:
Well that's exactly what fu-config.c does on the `fu_config_monitor_changed_cb`....

Reply:
In other news, you just discovered a nice memory leak in the existing code :) Want me to split up fu_config_load() for you?

Reply:
sure go ahead :)